frontend/lib/resolution/prims.cpp

```diff
@@ -511,6 +526,255 @@ static QualifiedType primObjectToInt(Context* context, const CallInfo& ci) {
+}
+
+static QualifiedType
+primIsNilableClassType(Context* context, const CallInfo& ci) {
+  if (ci.numActuals() < 1) return {};
+  bool eval = false;
+  if (auto t = ci.actual(0).type().type()) {
+    if (auto mt = t->toClassType()) eval = mt->decorator().isNilable();
+  }
+  return makeParamBool(context, eval);
+}
+
+static QualifiedType
+primIsNonNilableClassType(Context* context, const CallInfo& ci) {
+  if (ci.numActuals() < 1) return {};
+  bool eval = false;
+  if (auto t = ci.actual(0).type().type()) {
+    if (auto mt = t->toClassType()) eval = mt->decorator().isNonNilable();
+  }
+  return makeParamBool(context, eval);
+}
+
+static QualifiedType primIsRecordType(Context* context, const CallInfo& ci) {
+  if (ci.numActuals() < 1) return {};
+  bool eval = false;
+  if (auto t = ci.actual(0).type().type()) eval = t->isUserRecordType();
+  return makeParamBool(context, eval);
+}
+
+static QualifiedType primIsFcfType(Context* context, const CallInfo& ci) {
+  CHPL_UNIMPL("PRIM_IS_FCF_TYPE");
+  return {};
+}
+
+static QualifiedType primIsUnionType(Context* context, const CallInfo& ci) {
+  if (ci.numActuals() < 1) return {};
+  bool eval = false;
+  if (auto t = ci.actual(0).type().type()) eval = t->isUnionType();
+  return makeParamBool(context, eval);
+}
+
+static QualifiedType primIsExternType(Context* context, const CallInfo& ci) {
+  CHPL_UNIMPL("PRIM_IS_EXTERN_TYPE");
+  return {};
+}
+
+static QualifiedType
+primIsExternUnionType(Context* context, const CallInfo& ci) {
+  auto qt1 = primIsExternType(context, ci);
+  auto qt2 = primIsUnionType(context, ci);
+  const bool eval = qt1.isParamTrue() && qt2.isParamTrue();
+  return makeParamBool(context, eval);
+}
+
+static QualifiedType
+primIsAtomicType(Context* context, const CallInfo& ci) {
+  if (ci.numActuals() < 1) return {};
+  bool eval = false;
+  if (auto t = ci.actual(0).type().type()) {
+    eval = t->hasPragma(context, PRAGMA_ATOMIC_TYPE);
+  }
+  return makeParamBool(context, eval);
+}
+
+static QualifiedType
+primIsRefIterType(Context* context, const CallInfo& ci) {
+  CHPL_UNIMPL("PRIM_IS_REF_ITER_TYPE");
+  return {};
+}
+
+static QualifiedType
+primIsBorrowedClassType(Context* context, const CallInfo& ci) {
+  if (ci.numActuals() < 1) return {};
+  bool eval = false;
+  if (auto t = ci.actual(0).type().type()) {
+    if (auto mt = t->toClassType()) eval = mt->decorator().isBorrowed();
+  }
+  return makeParamBool(context, eval);
+}
+
+static QualifiedType
+primIsAbsEnumType(Context* context, const CallInfo& ci) {
+  if (ci.numActuals() < 1) return {};
+  bool eval = false;
+  if (auto t = ci.actual(0).type().type()) {
+    if (auto et = t->toEnumType()) eval = et->isAbstract();
+  }
+  return makeParamBool(context, eval);
+}
+
+static QualifiedType
+primIsPod(Context* context, const CallInfo& ci) {
+  CHPL_UNIMPL("PRIM_IS_POD");
+  return {};
+}
+
+static QualifiedType
+primIsCoercible(Context* context, const CallInfo& ci) {
+  if (ci.numActuals() < 2) return {};
```

Comment:
Why not enforce (here and elsewhere) `== 2`? The production compiler doesn't care, but any uses that give more arguments are arguably improper and not specified by the spec (since they're primitives), so I think we have a right to break them.